### PR TITLE
Upgrade Azure Terraform provider to v2.x.x

### PR DESCRIPTION
**Pull request by fabianofranz**
Upgrading the vendored version of the Azure Terraform provider to v2.8.0 and updating TF templates that require changes.

### CODE REVIEW

**Comment by fabianofranz** [comment]
@abhinavdahiya @jhixson74 this is ready for review, PTAL. Tks!

**Comment by jhixson74** [comment]
This looks good to me. My only question here is has this been tested with IPv6 / multiple address prefixes? Do we even have a CI job for that?

/approve

**Comment by fabianofranz** [comment]
> My only question here is has this been tested with IPv6 / multiple address prefixes?

I didn't test it in that scenario, I only relied on CI and manually tested regular setups a few times. But I'd be happy to manually test IPv6/address_prefixes if you have the directions.  ;-)

**Comment by jhixson74** [comment]
Sure thing. Here is the config I used for testing this:
<pre>
apiVersion: v1
baseDomain: installer.azure.devcluster.openshift.com
compute:
- hyperthreading: Enabled
  name: worker
  platform: {}
  replicas: 3
controlPlane:
  hyperthreading: Enabled
  name: master
  platform: {}
  replicas: 3
metadata:
  creationTimestamp: null
  name: ip6
networking:
  clusterNetwork:
  - cidr: fd01::/48
    hostPrefix: 64
  machineNetwork:
  - cidr: 10.0.0.0/16
  - cidr: fc00::/48
  networkType: OVNKubernetes
  serviceNetwork:
  - fd02::/112
platform:
  azure:
    baseDomainResourceGroupName: os4-common
    region: centralus
publish: External
...
</pre>

When you run the installer, make sure that you set OPENSHIFT_INSTALL_AZURE_EMULATE_SINGLESTACK_IPV6=true is set.

File: data/data/azure/bootstrap/main.tf [hunk 1]
```diff
@@ -169,50 +169,44 @@ resource "azurerm_network_interface_backend_address_pool_association" "internal_
...
+    storage_account_uri = var.storage_account.primary_blob_endpoint
   }
 
-  boot_diagnostics {
-    enabled     = true
-    storage_uri = var.storage_account.primary_blob_endpoint
+  timeouts {
+    create = "60m"
+    update = "60m"
+    delete = "60m"
```
Comment (abhinavdahiya) [inline data/data/azure/bootstrap/main.tf:209]: why such long timeouts? what are the defaults?
https://github.com/terraform-providers/terraform-provider-azurerm/blob/00f95f01cdf55a53f3877364115903864933c864/azurerm/internal/services/compute/linux_virtual_machine_resource.go#L42-L47

^^ these defaults look good enough to me,

Comment (fabianofranz) [reply data/data/azure/bootstrap/main.tf]: Right, the defaults should be good enough. Removed.

**Comment by fabianofranz** [comment]
@abhinavdahiya @jhixson74 PR has been rebased, this should be good now that 4.6 has open.

File: data/data/azure/master/master.tf [hunk 2]
```diff
@@ -78,54 +78,49 @@ resource "azurerm_network_interface_backend_address_pool_association" "master_in
...
-  os_profile_linux_config {
-    disable_password_authentication = false
+  boot_diagnostics {
+    storage_account_uri = var.storage_account.primary_blob_endpoint
   }
 
-  boot_diagnostics {
-    enabled     = true
-    storage_uri = var.storage_account.primary_blob_endpoint
+  timeouts {
```
Comment (abhinavdahiya) [inline data/data/azure/master/master.tf:120]: I think you missed these @fabianofranz 
https://github.com/openshift/installer/pull/3526#discussion_r432078898

Comment (fabianofranz) [reply data/data/azure/master/master.tf]: Good catch, fixed.

**Pull request merged**